src/sentry/api/authentication.py

```diff
@@ -561,96 +561,91 @@ def authenticate_token(self, request: Request, token: str) -> tuple[Any, Any]:
-    """
-
-    if not shared_secret_setting:
-        raise RpcAuthenticationSetupException(
-            f"Cannot validate {service_name} RPC request signatures without shared secret"
-        )
-
-    # Ensure no empty secrets
-    if any(not secret.strip() for secret in shared_secret_setting):
-        raise RpcAuthenticationSetupException(
-            f"Cannot validate {service_name} RPC request signatures with empty shared secret"
-        )
-
-    if not signature.startswith("rpc0:"):
-        logger.error("%s RPC signature validation failed: invalid signature prefix", service_name)
-        return False
-
-    try:
-        # We aren't using the version bits currently.
-        _, signature_data = signature.split(":", 2)
-
-        signature_input = body
-
-        for key in shared_secret_setting:
-            computed = hmac.new(key.encode(), signature_input, hashlib.sha256).hexdigest()
-            is_valid = constant_time_compare(computed.encode(), signature_data.encode())
-            if is_valid:
-                return True
-    except Exception:
-        logger.exception("%s RPC signature validation failed", service_name)
-        return False
-
-    logger.error("%s RPC signature validation failed", service_name)
-
-    return False
-
-
 class ServiceRpcSignatureAuthentication(StandardAuthentication):
     """
-    Generic authentication for service RPC requests.
+    Generic authentication for service requests.
     Requests are sent with an HMAC signed by a shared private key.
 
     Subclasses should define:
+    - service_name: str - name of the service (e.g. "launchpad")
     - shared_secret_setting_name: str - name of the settings attribute (e.g., "SEER_RPC_SHARED_SECRET")
-    - service_name: str - name of the service for logging (e.g., "Seer", "Launchpad")
-    - sdk_tag_name: str - name for the SDK tag (e.g., "seer_rpc_auth", "launchpad_rpc_auth")
     """
 
     token_name = b"rpcsignature"
-    shared_secret_setting_name: str
-    service_name: str
-    sdk_tag_name: str
+
+    service_name: ClassVar[str]
+    shared_secret_setting_name: ClassVar[str]
 
     def accepts_auth(self, auth: list[bytes]) -> bool:
+        # auth should be [b"Rpcsignature", b"rpc0:<hex>"]
         if not auth or len(auth) < 2:
             return False
         return auth[0].lower() == self.token_name
 
     def authenticate_token(self, request: Request, token: str) -> tuple[Any, Any]:
-        shared_secret_setting = getattr(settings, self.shared_secret_setting_name, None)
+        if not self.compare_signature(request.path_info, request.body, token):
+            raise AuthenticationFailed("Invalid signature")
 
-        if shared_secret_setting is None:
+        scope = sentry_sdk.get_isolation_scope()
+        scope.set_tag("api_token_type", self.token_name)
+        scope.set_tag("service_auth", self.service_name)
+
+        return AnonymousUser(), AuthenticatedToken(kind="service", entity_id=self.service_name)
```

Comment:
OK if I'm following everything correctly:
- This is the core change. Previously we were returning `token: str`, and now we are returning a real `AuthenticatedToken` like you mentioned
- the other changes are largely refactoring `compare_signature` to be an instance method in this class + cleanup
- launchpad is currently the only subclass of `ServiceRpcSignatureAuthentication`, but other services should probably standardize on this as well

This looks reasonable to me, but I suggest wrapping the return value in a feature flag if possible so we don't have to wait for a revert to deploy if something goes wrong.